torchvision/transforms/functional.py

```diff
@@ -82,6 +82,54 @@ def to_tensor(pic):
         return img
 
 
+def as_tensor(pic):
+    """Convert a ``PIL Image`` or ``numpy.ndarray`` to tensor of same type.
+
+    See ``AsTensor`` for more details.
+
+    Args:
+        pic (PIL Image or numpy.ndarray): Image to be converted to tensor.
+
+    Returns:
+        Tensor: Converted image.
+    """
+    if not(_is_pil_image(pic) or _is_numpy(pic)):
+        raise TypeError('pic should be PIL Image or ndarray. Got {}'.format(type(pic)))
+
+    if _is_numpy(pic) and not _is_numpy_image(pic):
+        raise ValueError('pic should be 2/3 dimensional. Got {} dimensions.'.format(pic.ndim))
+
+    if isinstance(pic, np.ndarray):
+        # handle numpy array
+        if pic.ndim == 2:
+            pic = pic[:, :, None]
+
+        img = torch.from_numpy(pic.transpose((2, 0, 1)))
+        return img
```

Comment:
I'm wondering if we should even support handling `np.ndarray` in this function.

Indeed, the data in a `np.ndarray` can have any format (for example, it can be a float array with range from 0-255), and we can't properly handle all possible cases. It's the responsibility of the user to do it. Plus, if the user passes OpenCV arrays to the function, it will be in `BGR` format (different from RGB from Pillow and what we use in torchvision)

As such, I think that we should probably only handle PIL Images -- handling numpy arrays is trivial from the user perspective (`torch.as_tensor(ndarray)`)

Reply:
I'm not really for or against keeping the numpy conversion here.

With that said I think the primary purpose of this function is doing the conversion to a pytorch tensor format and making it into a channels first format.  The scope of the inputs can therefore be broadened or narrowed without really affecting the goals of the function.  

The OpenCV arrays will still come out to be channels first after being passed through this function.  We do not need to make any other assumptions other than the data format is HWC (or in the case of black&white images that it is HW). 

So let me know if you think it's best to drop numpy.

Reply:
I think what we should aim for is the least amount of potential user-errors or surprises.

Indeed, both OpenCV and `scikit-image` returns images as ndarrays of HWC format. But the color convention is not the same, and from our perspective there is no way to know if the array is indeed HWC or not (imagine multi-band images for example).
What scaries me is that the ndarray that is passed could also be CHW for some reason, and the function would just return something wrong.

For that reason, we could try to make the scope of this function to be as narrow as possible, so that we can be sure we won't be mishandling user inputs.
PIL Images and AccImage have a well-defined format representation (although I'm not sure that many people use AccImage actually), which is not the case for ndarrays which are generic data containers.

Let me know what you think

Reply:
I am okay with the narrowed scope and have made changes to reflect that.  

As a separate point I would like to keep the functionality of ToPILImage to accept numpy arrays. That way users can still load in numpy (or other formats that they can convert to numpy) and as long as the user can convert the numpy array to PIL Image the sequence of compose will still work (as written below).

```python
transforms.Compose([
    transforms.ToPILImage(),
    ...
    transforms.PILToTensor(),
 ])
```

Reply:
Sounds good, changing `ToPILImage` was not in the plans (it would be a backwards-incompatible change)